addons/im_livechat/static/src/embed/common/chat_window_patch.js

```diff
@@ -1,18 +1,41 @@
 import { FeedbackPanel } from "@im_livechat/embed/common/feedback_panel/feedback_panel";
+import { CloseConfirmation } from "@im_livechat/embed/common/close_confirmation";
 
 import { ChatWindow } from "@mail/core/common/chat_window";
 
-import { useState } from "@odoo/owl";
+import { useState, toRaw } from "@odoo/owl";
 
 import { useService } from "@web/core/utils/hooks";
 import { patch } from "@web/core/utils/patch";
 
-Object.assign(ChatWindow.components, { FeedbackPanel });
+Object.assign(ChatWindow.components, { FeedbackPanel, CloseConfirmation });
 
 patch(ChatWindow.prototype, {
     setup() {
         super.setup(...arguments);
         this.livechatService = useService("im_livechat.livechat");
         this.chatbotService = useState(useService("im_livechat.chatbot"));
+        this.livechatState = useState({
+            showCloseConfirmation: false,
+        });
+    },
+
+    async close() {
+        const chatWindow = toRaw(this.props.chatWindow);
+        if (!this.livechatState.showCloseConfirmation) {
+            if (chatWindow.folded) {
+                this.props.chatWindow.open({ notifyState: this.thread?.state !== "open" });
+            }
+            this.livechatState.showCloseConfirmation = true;
```

Comment:
A folded chat window for livechat visitor are not closeable, the the condition on `chatWindow.folded` is useless code.

Better have this kind of LOCs only if it's actually doable in practice, so that we can cover these LOCs with test. As it is, this is over-engineering that has no test coverage, so likely to contain bad and unused code.